### PR TITLE
HW6/MVVM

### PR DESCRIPTION
**Pull request by alexey-chaichuk**
Screenshot_1609074626

### CODE REVIEW

File: app/src/main/java/ru/chaichuk/hwapp/MainActivity.kt [hunk 1]
```diff
@@ -31,4 +37,10 @@ class MainActivity :
     override fun onDetailsBack() {
         supportFragmentManager.popBackStack()
     }
+
+    override suspend fun loadMoviesList(): List<Movie> =
```
Comment (Axrorxoja) [inline app/src/main/java/ru/chaichuk/hwapp/MainActivity.kt:41]: why need this un used method

Comment (Axrorxoja) [reply app/src/main/java/ru/chaichuk/hwapp/MainActivity.kt:41]: another case if this method usable 
need to move to viewmodel

File: app/src/main/java/ru/chaichuk/hwapp/viewModels/MovieDetailsViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import androidx.lifecycle.ViewModel
+import ru.chaichuk.hwapp.data.Movie
+
+class MovieDetailsViewModel : ViewModel() {
+
+    private val _mutableMovie = MutableLiveData<Movie>()
+
+    val movie: LiveData<Movie> get() = _mutableMovie
+
+    fun loadMovie (newMovie : Movie) {
```
Comment (Axrorxoja) [inline app/src/main/java/ru/chaichuk/hwapp/viewModels/MovieDetailsViewModel.kt:14]: very strange method 
why we need like so the method 
which problem it solve ?

File: app/src/main/java/ru/chaichuk/hwapp/viewModels/MoviesListViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.launch
+import ru.chaichuk.hwapp.data.Movie
+import ru.chaichuk.hwapp.data.MoviesListLoader
+
+class MoviesListViewModel : ViewModel(){
+
+    var moviesListLoader: MoviesListLoader? = null
```
Comment (Axrorxoja) [inline app/src/main/java/ru/chaichuk/hwapp/viewModels/MoviesListViewModel.kt:13]: why does the adapter need an inside ViewModel?
something strange logic

File: app/src/main/java/ru/chaichuk/hwapp/viewModels/MovieDetailsViewModel.kt (hunk 2 above)
Comment (Axrorxoja) [reply app/src/main/java/ru/chaichuk/hwapp/viewModels/MovieDetailsViewModel.kt:14]: where he used

File: app/src/main/java/ru/chaichuk/hwapp/viewModels/MoviesListViewModel.kt (hunk 3 above)
Comment (alexey-chaichuk) [reply app/src/main/java/ru/chaichuk/hwapp/viewModels/MoviesListViewModel.kt:13]: I had two options:
1) pass application or activity context to ViewModel, as loadMovies(context) needs it
2) pass external loader as adapter, to not pass context
As I heard in lecture, it is not advisable to pass context to VewModel (or use AndroidViewModel) for easier testing, so I decided to pass adapter. I've also tried to pass context, and tried variant with global static application context. But leaved final variant with adapter.
What is better variant to call loadMovies(context)?

File: app/src/main/java/ru/chaichuk/hwapp/viewModels/MovieDetailsViewModel.kt (hunk 2 above)
Comment (alexey-chaichuk) [reply app/src/main/java/ru/chaichuk/hwapp/viewModels/MovieDetailsViewModel.kt:14]: it is called in onViewCreated for MoviesDetailsFragment, and I pass to it movie data from arguments, which I got from main activity

File: app/src/main/java/ru/chaichuk/hwapp/MainActivity.kt (hunk 1 above)
Comment (alexey-chaichuk) [reply app/src/main/java/ru/chaichuk/hwapp/MainActivity.kt:41]: I tried not to use context in ViewModel, so I moved code that needs it to main activity

Comment (Axrorxoja) [reply app/src/main/java/ru/chaichuk/hwapp/MainActivity.kt:41]: okay I got you

**Review by Axrorxoja — APPROVED** [review]

**Pull request merged**